### PR TITLE
binloginfo: check grpc error message before return critical error.

### PR DESCRIPTION
**Pull request by coocood**
For grpc error `received message larger than max`, it's not a critical error,
we should not stop the server listener when we meet this error.

### CODE REVIEW

**Comment by coocood** [comment]
@shenli @WangXiangUSTC @GregoryIan PTAL

File: sessionctx/binloginfo/binloginfo.go [hunk 1]
```diff
@@ -89,6 +89,10 @@ func (info *BinlogInfo) WriteBinlog(clusterID uint64) error {
 		if err == nil {
 			return nil
 		}
+		if strings.Contains(err.Error(), "received message larger than max") {
```
Comment (shenli) [inline sessionctx/binloginfo/binloginfo.go]: Should we return a more friendly error message here? Such as "too large binlog xxxx".

**Comment by coocood** [comment]
@shenli PTAL

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**